### ISSUE TITLE
Scaling issue of down sampling

### ISSUE THREAD

**Issue by wonagram** [minhtriet/graddft-qnn#31]
Our down sampling occurs scaling issue. Look at the y-axis scale of the figures below. 

down sampling by 2**6
<img width="547" alt="Image" src="https://github.com/user-attachments/assets/768a186f-9c00-43b2-88c8-f11ad6aaee68" />

down sampling by 2*14
<img width="546" alt="Image" src="https://github.com/user-attachments/assets/93588882-8b42-4c8b-b435-24cb5cc59d72" />

Let me explain why we have scaling mismatch issue. 
For example, if the original n_grid is 102, we're gonna downsample it to 34.

Without downsampling,
```
coefficients = [c1, c2, c3, c4, ..., c100, c101, c102]
densities= [d1, d2, d3, d4, ... ,d100, d101, d102]
grid = [w1, w2, ... ,w100, w101, w102]
xc_energy_density = [c1 d1, c2 d2, c3 d3, ... ,c100 d100, c101 d101, c102 d102]
xc_energy = a1 b1 w1 + ... + a102 b102 w102
```

With downsampling,
```
coefficients_down = [c1 + c2 + c3, c4 + c5 + c6 , ... , c100 + c101 + c102]
#QNNFunctional.compute_slice_sums(unscaled_densities, indices)
densities_down = [d1 + d2 + d3, d4 + d5 + d6, ..., d100 + d101 + d102] 
grid_down = [w1 + w2 + w3, ... , w100 + w101 + w102]
xc_energy_density_down = [(c1+ c2+ c3)(d1+d2+d3), ... , (c100+c101+c102)(d100+d101+d102)]
xc_energy_down = (c1+ c2+ c3)(d1+d2+d3)(w1+w2+w3) + ... + (c100+c101+c102)(d100+d101+d102)(w100+w101+w102)
```

let's suppose K is number of new indicies. (For above example, $$K =34, |G_k|=3$$). We can mathematically write those as
$$E_{xc}= \sum_{i=1}^{K \ G_k} c_i d_i w_i$$
$$E_{xc, down} =\sum_{k=1}^K (\sum_{i \in G_k} c_i)(\sum_{i \in G_k} d_i)(\sum_{i \in G_k} w_i)$$.

$$E_{xc, down}$$ scale is bigger as $K$  decrease or $|G_k|$ increase. 
I argue that we need to divide it by $K^\alpha$. I think $K^2$ works well.

Modified code will be like below. 
```
xc_energy_density = jnp.einsum("rf,rf->r", coefficients, densities)
xc_energy_density = abs_clip(xc_energy_density, clip_cte)
xc_energy_down= self._integrate(xc_energy_density, grid_weights)#sungwon
new_indicies_size = len(grid.weights)//(2**n_qubits)
return xc_energy_down/new_indicies_size**2
```
